test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenHandlerTests.cs

```diff
@@ -58,28 +60,36 @@ public static TheoryData<JwtTheoryData> SegmentTheoryData()
 
             JwtTestData.InvalidRegExSegmentsData(theoryData);
             JwtTestData.InvalidNumberOfSegmentsData("IDX14110:", theoryData);
+            JwtTestData.InvalidEncodedSegmentsData("", theoryData);
+            JwtTestData.ValidEncodedSegmentsData(theoryData);
 
             return theoryData;
         }
 
         // Tests checks to make sure that the token string created by the JsonWebTokenHandler is consistent with the 
         // token string created by the JwtSecurityTokenHandler.
-        [Theory, MemberData(nameof(CreateTokenTheoryData))]
-        public void CreateToken(CreateTokenTheoryData theoryData)
+        [Theory, MemberData(nameof(CreateJWETheoryData))]
+        public void CreateJWE(CreateTokenTheoryData theoryData)
         {
-            var context = TestUtilities.WriteHeader($"{this}.CreateToken", theoryData);
+            var context = TestUtilities.WriteHeader($"{this}.CreateJWE", theoryData);
             try
             {
-                string jwtFromJwtHandler = theoryData.JwtSecurityTokenHandler.CreateEncodedJwt(theoryData.TokenDescriptor);
-                string jwtFromJsonHandler = theoryData.JsonWebTokenHandler.CreateToken(theoryData.Payload, KeyingMaterial.JsonWebKeyRsa256SigningCredentials);
+                string jweFromJwtHandler = theoryData.JwtSecurityTokenHandler.CreateEncodedJwt(theoryData.TokenDescriptor);
+                string jweFromJsonHandler = theoryData.JsonWebTokenHandler.CreateToken(theoryData.Payload, theoryData.TokenDescriptor.SigningCredentials, theoryData.TokenDescriptor.EncryptingCredentials);
 
-                theoryData.JwtSecurityTokenHandler.ValidateToken(jwtFromJwtHandler, theoryData.ValidationParameters, out SecurityToken validatedToken);
-                theoryData.JsonWebTokenHandler.ValidateToken(jwtFromJsonHandler, theoryData.ValidationParameters);
+                theoryData.JwtSecurityTokenHandler.ValidateToken(jweFromJwtHandler, theoryData.ValidationParameters, out SecurityToken validatedToken);
+                theoryData.JsonWebTokenHandler.ValidateToken(jweFromJsonHandler, theoryData.ValidationParameters);
 
                 theoryData.ExpectedException.ProcessNoException(context);
-                var jwtTokenFromBuilder = new JsonWebToken(jwtFromJwtHandler);
-                var jwtTokenFromHandler = new JsonWebToken(jwtFromJsonHandler);
-                IdentityComparer.AreEqual(jwtTokenFromBuilder, jwtTokenFromHandler, context);
+                var jweTokenFromJwtHandler = new JsonWebToken(jweFromJwtHandler);
```

Comment:
Can we add a test to compare at the Claims level.

Reply:
Added this in for all the tests.